pkg/hubble/parser/seven/parser.go

```diff
@@ -89,8 +89,8 @@ func (p *Parser) Decode(r *accesslog.LogRecord, decoded *flowpb.Flow) error {
 
 	ip := decodeIP(r.IPVersion, r.SourceEndpoint, r.DestinationEndpoint)
 
-	sourceIP := net.ParseIP(ip.Source)
-	destinationIP := net.ParseIP(ip.Destination)
+	sourceIP, _ := netip.ParseAddr(ip.Source)
+	destinationIP, _ := netip.ParseAddr(ip.Destination)
```

Comment:
Last time I was looking at this, I was considering just rejecting flows with invalid IPs (i.e. return an error here, if `ParseAddr` fails). https://github.com/cilium/cilium/issues/22949 indicates that flows with invalid addresses can occur, but it's not yet clear why.

Reply:
I agree and think we should check the `, ok` value here and return an error.

Reply:
@gandro @tklauser You mean only here or in all decoders? I did that in all decoders, but the looked at failing tests and reverted the change.

Currently L7 parsers don't require log records to include `SourceEndpoint` or `DestinationEndpoint` at all. Are there possible cases when they'd be entirely empty?

For L3/4 parser it's also a behaviour change - at the moment it doesn't error if it receives e.g. an Ethernet packet, without IP layer. I don't know how intentional it is, do you think the parser should return an error in such case? If so, I'd leave this change for a separate PR.

Reply:
Ah, good point regarding L3/L4. I think we can observe packets that are Ethernet-only (e.g. ARP) so we probably should not reject flows that do not have any IP address.

For L7, we don't know. Issue https://github.com/cilium/cilium/issues/22949 seems to indicate that those flows seem to be able to occur in practice, but it's not yet clear what kind of flows those are.

So I guess we probably should keep the existing behavior (and ignore `ok`) for now.

Reply:
Agree, let's keep the existing behavior (i.e. ignore `, ok`) for now and follow up in a separate PR if needed.

Reply:
Reading this, it seems like we should explicitly add a comment to why we're ignoring `, ok`, since multiple people were confused about it.

Reply:
ok, I added some comments